src/main/java/com/igot/cb/playlist/service/impl/PlayListServiceImpl.java

```diff
@@ -202,24 +207,29 @@ private void persistInRedis(ObjectNode enrichedContentJson, PlayListEntity jsonN
 
   }
 
-  private Map<String, Map<String, Object>> fetchContentDetails(JsonNode childrenNode) {
+  private Map<String, Object> fetchContentDetails(JsonNode childrenNode) {
     log.info("PlayListService::fetchContentDetails");
-    Map<String, Map<String, Object>> enrichContentMaps = new HashMap<>();
+    Map<String, Object> compositeSearchRes = new HashMap<>();
     if (childrenNode != null && childrenNode.isArray()) {
-      childrenNode.forEach(childNode -> {
-        String childId = childNode.asText();
-        // Call method to fetch details for each child
-        Map<String, Object> contentResponse = contentService.readContentFromCache(childId,
-            new ArrayList<>());
-        if (MapUtils.isNotEmpty(contentResponse)) {
-          if (Constants.LIVE.equalsIgnoreCase((String) contentResponse.get(Constants.STATUS))) {
-            enrichContentMaps.put(childId, contentResponse);
-          }
-        }
-      });
+      HashMap<String, Object> reqBody = new HashMap<>();
+      HashMap<String, Object> req = new HashMap<>();
+      req.put(Constants.FACETS, Arrays.asList(cbServerProperties.getCourseCategoryFacet()));
```

Comment:
Not sure why this is added?
couldn't find any place where you are using these facets in your response - is this used by client?
get the value from constants which has courseCategory

Reply:
Yes This is playList API @karthik-tarento where we want factes info

Reply:
@karthik-tarento we need facets in the playList APIs so added this changes.